`macos-latest` + `rustc 1.48` = ` error: linking with cc failed: exit code: 1`

The rust 1.48 build started failing on macOS in #67.

```
   Compiling proc-macro2 v1.0.40
   Compiling cc v1.0.73
   Compiling quote v1.0.20
     Running `rustc --crate-name build_script_build --edition=2018 /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.40/build.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="default"' --cfg 'feature="proc-macro"' -C metadata=a775773b1c4d6d13 -C extra-filename=-a775773b1c4d6d13 --out-dir /Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13 -L dependency=/Users/runner/work/minisafe/minisafe/target/debug/deps --cap-lints allow`
     Running `rustc --crate-name cc --edition=2018 /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/cc-1.0.73/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 -C metadata=6951f41806db0446 -C extra-filename=-6951f41806db0446 --out-dir /Users/runner/work/minisafe/minisafe/target/debug/deps -L dependency=/Users/runner/work/minisafe/minisafe/target/debug/deps --cap-lints allow`
     Running `rustc --crate-name build_script_build --edition=2018 /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/quote-1.0.20/build.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="default"' --cfg 'feature="proc-macro"' -C metadata=43d0865adc1d5c42 -C extra-filename=-43d0865adc1d5c42 --out-dir /Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42 -L dependency=/Users/runner/work/minisafe/minisafe/target/debug/deps --cap-lints allow`
   Compiling unicode-ident v1.0.2
     Running `rustc --crate-name unicode_ident --edition=2018 /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/unicode-ident-1.0.2/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 -C metadata=830f4edd07508c85 -C extra-filename=-830f4edd07508c85 --out-dir /Users/runner/work/minisafe/minisafe/target/debug/deps -L dependency=/Users/runner/work/minisafe/minisafe/target/debug/deps --cap-lints allow`
   Compiling syn v1.0.98
     Running `rustc --crate-name build_script_build --edition=2018 /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/syn-1.0.98/build.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="clone-impls"' --cfg 'feature="default"' --cfg 'feature="derive"' --cfg 'feature="parsing"' --cfg 'feature="printing"' --cfg 'feature="proc-macro"' --cfg 'feature="quote"' -C metadata=f23be10b00cb500d -C extra-filename=-f23be10b00cb500d --out-dir /Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d -L dependency=/Users/runner/work/minisafe/minisafe/target/debug/deps --cap-lints allow`
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-m64" "-arch" "x86_64" "-L" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.0.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.1.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.10.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.11.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.12.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.13.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.14.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.15.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.2.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.3.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.4.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.5.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.6.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.7.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.8.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.build_script_build.ac18yd7a-cgu.9.rcgu.o" "-o" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42" "/Users/runner/work/minisafe/minisafe/target/debug/build/quote-43d0865adc1d5c42/build_script_build-43d0865adc1d5c42.2vq3dqw043l0ru7x.rcgu.o" "-Wl,-dead_strip" "-nodefaultlibs" "-L" "/Users/runner/work/minisafe/minisafe/target/debug/deps" "-L" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libstd-688c1376a25c049d.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-078f89c2cdb6d46b.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libobject-fec020208bc1ad3c.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libaddr2line-38540dcacc9fd218.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libgimli-6f613179f618c598.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_demangle-4d1a5d7118aaeaf2.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libhashbrown-d88fab3b1b9d8356.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_alloc-a21754532a052f2f.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libunwind-2e9ebc4127641a96.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcfg_if-82d0f961232a05ca.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liblibc-370b1b71f08bac3c.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liballoc-816106c1f35f5421.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_core-20f3a030f1a56a86.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcore-60333aa00936c5ce.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-116aaeaea873ef94.rlib" "-lSystem" "-lresolv" "-lc" "-lm"
  = note: ld: in /Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-078f89c2cdb6d46b.rlib(lib.rmeta), archive member 'lib.rmeta' with length 29624 is not mach-o or llvm bitcode file '/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-078f89c2cdb6d46b.rlib'
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-m64" "-arch" "x86_64" "-L" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.0.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.1.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.10.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.11.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.12.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.13.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.14.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.15.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.2.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.3.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.4.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.5.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.6.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.7.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.8.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.build_script_build.8y8agsyp-cgu.9.rcgu.o" "-o" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d" "/Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d/build_script_build-f23be10b00cb500d.3iv5l76q37nbdd9h.rcgu.o" "-Wl,-dead_strip" "-nodefaultlibs" "-L" "/Users/runner/work/minisafe/minisafe/target/debug/deps" "-L" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libstd-688c1376a25c049d.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-078f89c2cdb6d46b.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libobject-fec020208bc1ad3c.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libaddr2line-38540dcacc9fd218.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libgimli-6f613179f618c598.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_demangle-4d1a5d7118aaeaf2.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libhashbrown-d88fab3b1b9d8356.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_alloc-a21754532a052f2f.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libunwind-2e9ebc4127641a96.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcfg_if-82d0f961232a05ca.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liblibc-370b1b71f08bac3c.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liballoc-816106c1f35f5421.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_core-20f3a030f1a56a86.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcore-60333aa00936c5ce.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-116aaeaea873ef94.rlib" "-lSystem" "-lresolv" "-lc" "-lm"
  = note: ld: in /Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-078f89c2cdb6d46b.rlib(lib.rmeta), archive member 'lib.rmeta' with length 29624 is not mach-o or llvm bitcode file '/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-078f89c2cdb6d46b.rlib'
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: aborting due to previous error

error: aborting due to previous error

error: could not compile `syn`

Caused by:
  process didn't exit successfully: `rustc --crate-name build_script_build --edition=2018 /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/syn-1.0.98/build.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="clone-impls"' --cfg 'feature="default"' --cfg 'feature="derive"' --cfg 'feature="parsing"' --cfg 'feature="printing"' --cfg 'feature="proc-macro"' --cfg 'feature="quote"' -C metadata=f23be10b00cb500d -C extra-filename=-f23be10b00cb500d --out-dir /Users/runner/work/minisafe/minisafe/target/debug/build/syn-f23be10b00cb500d -L dependency=/Users/runner/work/minisafe/minisafe/target/debug/deps --cap-lints allow` (exit code: 1)
warning: build failed, waiting for other jobs to finish...
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-m64" "-arch" "x86_64" "-L" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.0.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.1.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.10.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.11.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.12.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.13.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.14.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.15.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.2.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.3.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.4.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.5.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.6.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.7.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.8.rcgu.o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.build_script_build.6v3bvynn-cgu.9.rcgu.o" "-o" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13" "/Users/runner/work/minisafe/minisafe/target/debug/build/proc-macro2-a775773b1c4d6d13/build_script_build-a775773b1c4d6d13.3mcole49nmerkt58.rcgu.o" "-Wl,-dead_strip" "-nodefaultlibs" "-L" "/Users/runner/work/minisafe/minisafe/target/debug/deps" "-L" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libstd-688c1376a25c049d.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-078f89c2cdb6d46b.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libobject-fec020208bc1ad3c.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libaddr2line-38540dcacc9fd218.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libgimli-6f613179f618c598.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_demangle-4d1a5d7118aaeaf2.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libhashbrown-d88fab3b1b9d8356.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_alloc-a21754532a052f2f.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libunwind-2e9ebc4127641a96.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcfg_if-82d0f961232a05ca.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liblibc-370b1b71f08bac3c.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liballoc-816106c1f35f5421.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_core-20f3a030f1a56a86.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcore-60333aa00936c5ce.rlib" "/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-116aaeaea873ef94.rlib" "-lSystem" "-lresolv" "-lc" "-lm"
  = note: ld: in /Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-078f89c2cdb6d46b.rlib(lib.rmeta), archive member 'lib.rmeta' with length 29624 is not mach-o or llvm bitcode file '/Users/runner/.rustup/toolchains/1.48-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-078f89c2cdb6d46b.rlib'
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: aborting due to previous error

error: build failed
Error: Process completed with exit code 101.
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -9,7 +9,7 @@ jobs:
       - uses: actions/checkout@v1
       - uses: actions-rs/toolchain@v1
         with:
-            toolchain: 1.67.1
+            toolchain: 1.70.0
             components: rustfmt, clippy
             override: true
       - name: rustfmt

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -22,7 +22,7 @@ jobs:
     strategy:
       matrix:
         toolchain:
-          - 1.48
+          - 1.54
           - nightly
         os:
           - ubuntu-latest
@@ -42,8 +42,7 @@ jobs:
         if: matrix.os == 'windows-latest'
         run: cargo test --verbose --no-default-features
       - name: Test on Rust ${{ matrix.toolchain }} (non Windows)
-        # See issue https://github.com/wizardsardine/liana/issues/69
-        if: matrix.os != 'windows-latest' && (matrix.os != 'macOS-latest' || matrix.toolchain != '1.48')
+        if: matrix.os != 'windows-latest'
         run: cargo test --verbose --color always -- --nocapture
 
   linter_gui:

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -66,7 +66,11 @@ query).
 
 ## Minimum Supported Rust Version
 
-`lianad` should always compile and pass tests using **Rust 1.48**.
+`lianad` should always compile and pass tests using **Rust 1.54**. The rationale behind this is
+support something reasonable, and preferably supported by all of:
+- [Guix](https://guix.gnu.org/)
+- Popular distributions' packages (especially Debian which is the most conservative)
+- [Mrustc](https://github.com/thepowersgang/mrustc)
 
 ## Style
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,9 +44,9 @@ fern = "0.6"
 # to work with our custom panic hook.
 backtrace = "0.3"
 
-# Pinned to this version because they broke the MSRV in 0.27...
+# Pinned to this version because they keep breaking their MSRV in point releases...
 # FIXME: this is unfortunate, we don't receive the updates (sometimes critical) from SQLite.
-rusqlite = { version = "0.26.3", features = ["bundled", "unlock_notify"] }
+rusqlite = { version = "0.27", features = ["bundled", "unlock_notify"] }
 
 # To talk to bitcoind
 jsonrpc = "0.12"

--- a/doc/BUILD.md
+++ b/doc/BUILD.md
@@ -8,10 +8,7 @@ this repository. They are pinned in a [`Cargo.lock`](../Cargo.lock) file at the 
 We take security very seriously, and toolchain is a big part of that. We are moderatly conservative
 with dependencies and aim to target reasonable compiler versions that have had time to mature (ie
 that had the chance to be reviewed and distributed by third parties, as well as tested by the
-community).  The minimum supported Rust version for `lianad` currently is `1.48`, that is the
-version of [`rustc` shipped in Debian stable](https://packages.debian.org/stable/rustc). (It is also
-inferior to the latest version of `rustc` supported by
-[`mrustc`](https://github.com/thepowersgang/mrustc/) at the time of writing, `1.54`).
+community).  The minimum supported Rust version for `lianad` currently is `1.54`.
 
 If you want to not only build the daemon but the whole wallet including the GUI, you'll
 unfortunately have to use a more recent `cargo`. The minimum version supported by the GUI at the

--- a/src/bitcoin/d/mod.rs
+++ b/src/bitcoin/d/mod.rs
@@ -1,6 +1,7 @@
-///! Implementation of the Bitcoin interface using bitcoind.
-///!
-///! We use the RPC interface and a watchonly descriptor wallet.
+//! Implementation of the Bitcoin interface using bitcoind.
+//!
+//! We use the RPC interface and a watchonly descriptor wallet.
+
 mod utils;
 use crate::{
     bitcoin::{Block, BlockChainTip},

--- a/src/bitcoin/mod.rs
+++ b/src/bitcoin/mod.rs
@@ -1,6 +1,7 @@
-///! Interface to the Bitcoin network.
-///!
-///! Broadcast transactions, poll for new unspent coins, gather fee estimates.
+//! Interface to the Bitcoin network.
+//!
+//! Broadcast transactions, poll for new unspent coins, gather fee estimates.
+
 pub mod d;
 pub mod poller;
 

--- a/src/commands/mod.rs
+++ b/src/commands/mod.rs
@@ -293,12 +293,10 @@ impl DaemonControl {
     /// Get a list of all known coins.
     pub fn list_coins(&self) -> ListCoinsResult {
         let mut db_conn = self.db.connection();
-        #[allow(clippy::iter_kv_map)] // Because Rust 1.48
         let coins: Vec<ListCoinsEntry> = db_conn
             .coins(CoinType::All)
-            // Can't use into_values as of Rust 1.48
-            .into_iter()
-            .map(|(_, coin)| {
+            .into_values()
+            .map(|coin| {
                 let Coin {
                     amount,
                     outpoint,
@@ -728,15 +726,12 @@ impl DaemonControl {
         let timelock =
             timelock.unwrap_or_else(|| self.config.main_descriptor.first_timelock_value());
         let height_delta: i32 = timelock.try_into().expect("Must fit, it's a u16");
-        let sweepable_coins = db_conn
-            .coins(CoinType::Unspent)
-            .into_iter()
-            .filter(|(_, c)| {
-                // We are interested in coins available at the *next* block
-                c.block_info
-                    .map(|b| current_height + 1 >= b.height + height_delta)
-                    .unwrap_or(false)
-            });
+        let sweepable_coins = db_conn.coins(CoinType::Unspent).into_values().filter(|c| {
+            // We are interested in coins available at the *next* block
+            c.block_info
+                .map(|b| current_height + 1 >= b.height + height_delta)
+                .unwrap_or(false)
+        });
 
         // Fill-in the transaction inputs and PSBT inputs information. Record the value
         // that is fed to the transaction while doing so, to compute the fees afterward.
@@ -744,7 +739,7 @@ impl DaemonControl {
         let txin_sat_vb = self.config.main_descriptor.max_sat_vbytes();
         let mut sat_vb = 0;
         let mut spent_txs = HashMap::new();
-        for (_, coin) in sweepable_coins {
+        for coin in sweepable_coins {
             in_value += coin.amount;
             psbt.unsigned_tx.input.push(bitcoin::TxIn {
                 previous_output: coin.outpoint,

--- a/src/database/mod.rs
+++ b/src/database/mod.rs
@@ -1,6 +1,7 @@
-///! Database interface for Liana.
-///!
-///! Record wallet metadata, spent and unspent coins, ongoing transactions.
+//! Database interface for Liana.
+//!
+//! Record wallet metadata, spent and unspent coins, ongoing transactions.
+
 pub mod sqlite;
 
 use crate::{

--- a/src/database/sqlite/mod.rs
+++ b/src/database/sqlite/mod.rs
@@ -1,11 +1,12 @@
-///! Implementation of the database interface using SQLite.
-///!
-///! We use a bundled SQLite that is compiled with SQLITE_THREADSAFE. Sqlite.org states:
-///! > Multi-thread. In this mode, SQLite can be safely used by multiple threads provided that
-///! > no single database connection is used simultaneously in two or more threads.
-///!
-///! We leverage SQLite's `unlock_notify` feature to synchronize writes accross connection. More
-///! about it at https://sqlite.org/unlock_notify.html.
+//! Implementation of the database interface using SQLite.
+//!
+//! We use a bundled SQLite that is compiled with SQLITE_THREADSAFE. Sqlite.org states:
+//! > Multi-thread. In this mode, SQLite can be safely used by multiple threads provided that
+//! > no single database connection is used simultaneously in two or more threads.
+//!
+//! We leverage SQLite's `unlock_notify` feature to synchronize writes accross connection. More
+//! about it at https://sqlite.org/unlock_notify.html.
+
 pub mod schema;
 mod utils;
 

